pattern/variable/SystemReference.java

```diff
@@ -18,17 +18,43 @@
 
 package grakn.core.pattern.variable;
 
-import graql.lang.common.GraqlToken;
 import graql.lang.pattern.variable.Reference;
 
+import java.util.Objects;
+
 public class SystemReference extends Reference.Name {
 
-    public SystemReference(final String name) {
-        super(name);
+    public static String PREFIX = "$/";
+
+    private final int id;
+    private final int hash;
+
+    public SystemReference(int id) {
+        super("sys" + id); // TODO: why are we we doing this? Why does this class have to extend Reference.Name?
+        this.id = id;
+        this.hash = Objects.hash(SystemReference.class, id);
+    }
```

Comment:
Why are we doing this? Why does this class have to extend Reference.Name? Does it ever need to be referred to as a "named variable"? I don't see that usage anywhere. If not then it should not subclass `Reference.Name`, and thus it should not need to prefix its `id` with a special string, such as "sys" (this was previously "temp").

Reply:
I think there's a case to make `System` a 4th type along with `Name`, `Label` and `Anonymous`. I'll briefly discuss this with Josh since we wrote that class together, but I don't think it strictly needs to be `Name`. It does however need to be Referreable.